cloudinstall/utils.py

```diff
@@ -101,6 +102,41 @@ def write_status_file(status='', msg=''):
     spew(status_file, json.dumps(dict(status=status, msg=msg)))
 
 
+def populate_config(opts):
+    """ populate configuration suitable for loading in the config
+    object merging in cli options.
+
+    :param opts: argparse Namespace class of options
+    """
+    cfg_cli_opts = vars(opts)
+    cfg = {}
+
+    def sanitize_config_items(_cfg):
+        """ remove false and null items """
+        return {k: v for (k, v) in _cfg.items()
+                if v is not None and v is not False}
```

Comment:
I dont think it's b/c of Config.getattr. One of the issues I ran into was if I have something like `headless: true` in the config yaml and I don't pass the `--headless` option to the script then the generated config object will automatically overwrite `headless: true` even though it shouldnt be altered.